api/handlers/capturesource.go

```diff
@@ -31,46 +31,184 @@ LICENSE
+func FromCaptureSource(captureSource *model.CaptureSource, id int, format *api.Format) CaptureSourceResult {
+	var result CaptureSourceResult
+	if format.Requires("id") {
+		result.ID = &id
+	}
+	if format.Requires("name") {
+		result.Name = &captureSource.Name
+	}
+	if format.Requires("location") {
+		location := fmt.Sprintf("%f,%f", captureSource.Location.Lat, captureSource.Location.Lng)
+		result.Location = &location
+	}
+	if format.Requires("camera_hardware") {
+		result.CameraHardware = &captureSource.CameraHardware
+	}
+	if format.Requires("site_id") {
+		result.SiteID = captureSource.SiteID
+	}
+	return result
+}
+
+// GetCaptureSourcesQuery describes the URL query parameters required for the GetCaptureSources endpoint.
+type GetCaptureSourcesQuery struct {
+	Name     *string `query:"name"`     // Optional.
+	Location *string `query:"location"` // Optional.
+	api.LimitAndOffset
+	// TODO: Code could be simplified if api.Format could be embedded here. Testing shows it cannot, reason unknown.
+}
+
+// CreateCaptureSourceBody describes the JSON format required for the CreateCaptureSource endpoint.
+// ID is omitted because it is chosen automatically. All other fields are required.
+type CreateCaptureSourceBody struct {
+	Name           string `json:"name"`
+	Location       string `json:"location"`
+	CameraHardware string `json:"camera_hardware"`
+	SiteID         *int64 `json:"site_id"` // Optional.
+}
+
+// GetCaptureSourceByID gets a capture source when provided with an ID.
 func GetCaptureSourceByID(ctx *fiber.Ctx) error {
-	// TODO: implement handler
+	// Parse URL.
+	format := new(api.Format)
+
+	if err := ctx.QueryParser(format); err != nil {
+		return api.InvalidRequestURL(ctx)
+	}
 
-	// id, _ := ctx.ParamsInt("id", 1)
-	return ctx.JSON("TODO")
+	id, err := strconv.ParseInt(ctx.Params("id"), 10, 64)
+	if err != nil {
+		return api.InvalidRequestURL(ctx)
+	}
+
+	// Fetch data from the datastore.
+	store := ds_client.Get()
+	key := store.IDKey("CaptureSource", id)
+	var captureSource model.CaptureSource
+	if store.Get(context.Background(), key, &captureSource) != nil {
+		return api.DatastoreReadFailure(ctx)
+	}
+
+	// Format result.
+	result := FromCaptureSource(&captureSource, int(id), format)
+
+	return ctx.JSON(result)
 }
 
+// GetCaptureSources gets a list of capture sources, filtering by name, location if specified.
 func GetCaptureSources(ctx *fiber.Ctx) error {
 	// Parse URL.
-	name := ctx.Query("name")
-	location := ctx.Query("location")
-	format := utils.GetFormat(ctx)
-	limit, offset := utils.GetLimitAndOffset(ctx, 20)
-
-	// Debugging info.
-	fmt.Println(name, location, format, limit, offset)
-
-	// Placeholder code: returns an empty result.
-	// TODO: implement fetching from datastore.
-	result := utils.Result[CaptureSourceResult]{
-		Results: []CaptureSourceResult{},
-		Offset:  offset,
-		Limit:   limit,
-		Total:   0,
+	qry := new(GetCaptureSourcesQuery)
```

Comment:
Would it be better to write a NewGetCaptureSourcesQuery function to provide a new one of these, which also does the limit set below ?